mr-plonky2-circuits/src/query_erc20/storage/tests.rs

```diff
@@ -25,393 +34,162 @@ use plonky2::{
-    /// Writes the parts of the public inputs into the provided target array.
-    pub fn parts_into_values(
-        values: &mut [T; PublicInputs::<()>::TOTAL_LEN],
-        root: &[T; PublicInputs::<()>::ROOT_LEN],
-        digest: &[T; PublicInputs::<()>::DIGEST_LEN],
-        owner: &[T; PublicInputs::<()>::OWNER_LEN],
-    ) {
-        values[Self::ROOT_OFFSET..Self::ROOT_OFFSET + Self::ROOT_LEN].copy_from_slice(root);
-        values[Self::DIGEST_OFFSET..Self::DIGEST_OFFSET + Self::DIGEST_LEN].copy_from_slice(digest);
-        values[Self::OWNER_OFFSET..Self::OWNER_OFFSET + Self::OWNER_LEN].copy_from_slice(owner);
-    }
-}
-
-impl<'a> PublicInputs<'a, GoldilocksField> {
-    /// Given a seed, generate & prove a leaf circuit, returning it along the value it encodes
-    /// meaning the mapping key
-    pub fn inputs_from_seed(
-        seed: u64,
-    ) -> (
-        [u8; MAPPING_KEY_LEN],
-        [GoldilocksField; PublicInputs::<GoldilocksField>::TOTAL_LEN],
-    ) {
-        Self::inputs_from_seed_and_owner(seed, Address::random())
-    }
-
-    pub(crate) fn inputs_from_seed_and_owner(
-        seed: u64,
-        owner_addr: Address,
-    ) -> (
-        [u8; MAPPING_KEY_LEN],
-        [GoldilocksField; PublicInputs::<GoldilocksField>::TOTAL_LEN],
-    ) {
-        let mut pis = [GoldilocksField::ZERO; PublicInputs::<GoldilocksField>::TOTAL_LEN];
-        let rng = &mut StdRng::seed_from_u64(seed);
-        // generate a fake NFT ID within u32 and in big endian encoding
-        let leaf_key = std::iter::repeat(0)
-            .take(MAPPING_KEY_LEN - 4)
-            .chain(rng.next_u32().to_be_bytes().into_iter())
-            .collect::<Vec<_>>();
-        let packed_leaf = convert_u8_to_u32_slice(&leaf_key);
-        let packed_leaf_f: [GoldilocksField; PACKED_MAPPING_KEY_LEN] =
-            array::from_fn(|i| F::from_canonical_u32(packed_leaf[i]));
-
-        // leaf value == owner address
-        let user_address = convert_u8_to_u32_slice(&left_pad32(owner_addr.as_fixed_bytes()));
-        let leaf_value: [GoldilocksField; PACKED_VALUE_LEN] =
-            array::from_fn(|i| F::from_canonical_u32(user_address[i]));
-
-        let root = F::rand_vec(NUM_HASH_OUT_ELTS).try_into().unwrap();
-        let digest = map_to_curve_point::<F>(&packed_leaf_f).to_weierstrass();
-        let digest_fs = digest
-            .x
-            .0
-            .iter()
-            .chain(digest.y.0.iter())
-            .copied()
-            .chain(std::iter::once(GoldilocksField::from_bool(digest.is_inf)))
-            .collect::<Vec<_>>();
-
-        Self::parts_into_values(
-            &mut pis,
-            &root,
-            digest_fs.as_slice().try_into().unwrap(),
-            &leaf_value,
-        );
+    let proof = run_circuit::<_, D, C, _>(test_circuit);
+    let [pi, child_pi] = [&proof.public_inputs, child_pi_slice]
+        .map(|pi| PublicInputs::<GoldilocksField>::from_slice(pi));
+    // TODO: check the hash (p[C])
```

Comment:
can you do it ? would be nice to ensure that the database leaf hash is still the same

Reply:
Added in commit [bc67b19](https://github.com/Lagrange-Labs/Euclid-database/pull/6/commits/bc67b192a05d040164a52a8ce9ba9bdcf952a573).